addons/crm/models/crm_activity.py

```diff
@@ -48,8 +48,9 @@ def create(self, values):
 
     @api.multi
     def unlink(self):
-        activities = self.search([('subtype_id', '=', self.subtype_id.id)])
-        # to ensure that the subtype is only linked the current activity
-        if len(activities) == 1:
-            self.subtype_id.unlink()
+        for record in self:
+            activities = self.search([('subtype_id', '=', record.subtype_id.id)])
```

Comment:
could you replaced by a `mapped` to make one search for all subtypes?

Reply:
Hey @mart-e. Do you think the `if len(activities) == 1` is necessary? I'm not sure why it's there and from the previous commit made on this file it is also not clear. Could we not simply do `self.env['mail.message.subtype'].search([('id', 'in', self.mapped(lambda r: r.subtype_id.id))]).unlink()`?

Reply:
executing a method on 0 record is usually not advised but feel free to submit something else as long as it is efficient :)

Reply:
@mart-e okay pull request has been updated